apps/science/main.cpp

```diff
@@ -1,13 +1,164 @@
 #include "mbed.h"
+#include "mbed_config.h"
+#include "can_config.h"
+
+#include "ScienceConfig.h"
+#include "Encoder.h"
+#include "EncoderAbsolute_PWM.h"
+#include "EncoderRelative_Quadrature.h"
+#include "ActuatorController.h"
+#include "CANMsg.h"
+#include "CANBuffer.h"
+#include "Servo.h"
+#include "MoistureSensor.h"
+
+//motors
+Motor indexerMotor(MTR_PWM_1, MTR_DIR_1, false);
+Motor elevatorMotor(MTR_PWM_2, MTR_DIR_2, false);
+
+//Servo
+Servo coverServo(SRVO_PWM_1, ScienceConfig::coverServoType, ScienceConfig::coverServoRange, ScienceConfig::coverServoMaxPulse, ScienceConfig::coverServoMinPulse);
+Servo diggerServo(SRVO_PWM_2, ScienceConfig::diggerServoType, ScienceConfig::diggerServoRange, ScienceConfig::diggerServoMaxPulse, ScienceConfig::diggerServoMinPulse);
+
+//encoders
+EncoderAbsolute_PWM elevatorEncoder(ScienceConfig::elevatorEncoderConfig);
+EncoderRelative_Quadrature indexerEncoder(ScienceConfig::centrifugeEncoderConfig);
+//limit switches
+
+DigitalIn indexerLimFront(LIM_SW_1);
+DigitalIn indexerLimBack(LIM_SW_2);
+DigitalIn elevatorLimTop(LIM_SW_3);
+DigitalIn elevatorLimBottom(LIM_SW_4);
+
+// Actuators
+ActuatorController indexerActuator(ScienceConfig::indexerActuatorConfig, indexerMotor, indexerEncoder, indexerLimFront, indexerLimBack);
+ActuatorController elevatorActuator(ScienceConfig::diggerLiftActuatorConfig, elevatorMotor, elevatorEncoder, elevatorLimBottom, elevatorLimTop);
+
+//I2C
+
+MoistureSensor sensor = MoistureSensor(TEMP_MOIST_I2C_SDA, TEMP_MOIST_I2C_SCL);
```

Comment:
I'm not sure if we'll be adding more sensors to the science unit in the future so you might want to name this sensor "MoistureSensor" or something more specific to this sensor.